populate_dev_data_handler.go

```diff
@@ -23,42 +23,49 @@ import (
     "appengine/datastore"
 )
 
+// Create TestRun entities for local development and testing.
+// These point to JSON files stored in /static/.
+// Key names for each entity are specified to make this route idempotent.
 func populateDevData(w http.ResponseWriter, r *http.Request) {
     ctx := appengine.NewContext(r)
```

Comment:
I think contexts are used one-per-request in Go because the language doesn't support traditional OOP.

I found this explanation which has slightly more context (pun very much intended:1st_place_medal: :boom:)

> At Google, we developed a context package that makes it easy to pass request-scoped values, cancelation signals, and deadlines across API boundaries to all the goroutines involved in handling a request.

https://blog.golang.org/context